core/services/relay/evm/mercury/wsrpc/client.go

```diff
@@ -298,10 +298,18 @@ func (w *client) handleTimeout(err error) {
 				// Debug log in case my reasoning is wrong.
 				w.logger.Debugf("Transport is resetting, cnt=%d", cnt)
 			}
+			return
 		}
 	} else {
 		w.consecutiveTimeoutCnt.Store(0)
 	}
+
+	// if the underlying client is closed by another
+	state := w.conn.GetState()
+	if state != connectivity.Ready && state != connectivity.Connecting {
+		w.logger.Warnw("Transport is resetting - connection state unusable", "state", w.conn.GetState())
+		w.chResetTransport <- struct{}{}
+	}
```

Comment:
How would this happen out of interest? The underlying client should be exclusive to this one.